packages/plugin/src/generators/app/templates/.storybook/withMockApi.tsx

```diff
@@ -0,0 +1,49 @@
+import { http, HttpResponse } from 'msw';
+import React, { useEffect, useState } from 'react';
+
+export type MockDefinition = {
+    method: 'GET' | 'POST' | 'PUT' | 'PATCH' | 'DELETE';
+    endpoint: string;
+    response: Record<string, unknown>;
+    status?: number;
+    delayMs?: number;
+};
+
+export type WithApiProps = {
+    mocks: MockDefinition[];
+    children: React.ReactElement;
+};
+
+export const WithMockApi = ({ mocks, children }: WithApiProps) => {
+    const [key, setKey] = useState('initial');
+
+    useEffect(() => {
+        const worker = (window as any).msw?.worker;
+        if (!worker) {
+            console.warn('[WithMockAPI] MSW worker not found on window');
+            return;
+        }
+
+        const handlers = mocks.map((mock) => {
+            const fullUrl = `${window.location.origin}${mock.endpoint}`;
+            const method = mock.method.toLowerCase() as keyof typeof http;
+
+            return http[method](fullUrl, async () => {
+                if (mock.delayMs) await new Promise((res) => setTimeout(res, mock.delayMs));
+                return HttpResponse.json(mock.response, { status: mock.status ?? 200 });
+            });
+        });
+
+        console.log('[WithMockAPI] Registering handlers:', mocks);
+        worker.resetHandlers(...handlers);
+    }, [JSON.stringify(mocks)]);
+
+    useEffect(() => {
+        const newKey = mocks
+            .map((m) => `${m.method}-${m.endpoint}-${m.status}-${JSON.stringify(m.response)}`)
+            .join('|');
+        setKey(newKey);
```

Comment:
Is this to force remounting the widget when the mocks change?  Could probably just set key to any random value, or the next value in an incrementing sequence if you don't want to add any more dependencies.